### PR TITLE
Make it possible to select or deselect test groups by number

### PR DESCRIPTION
**Pull request by levitte**
Examples of possible expressions (adapt to your platform):

    make test TESTS=-99
    make test TESTS=10
    make test TESTS=-9?
    make test TESTS=-[89]0
    make test TESTS=[89]0

### CODE REVIEW

**Comment by richsalz** [comment]
Having looked at the diff, I don't understand what a leading minus sign is supposed to do.

**Comment by levitte** [comment]
`TESTS=-test_fuzz` runs all tests except `99-test_fuzz`.
`TESTS=-99` runs all tests except those in the 99 group. 
`TESTS='-9?'` runs all tests except those in groups 90 to 99.
`TESTS='8? -90'` runs all tests in groups 80 to 99 except those in group 90.

**Comment by richsalz** [comment]
But I don't see how the minus sign works, unless I am missing some special glob magic.

**Comment by levitte** [comment]
Ah, that's detected [here](https://github.com/openssl/openssl/pull/2887/files#diff-74c0e843b39ff68490f8a58e7ded6379R55). The finding function never gets to see it.

**Comment by kaduk** [comment]
(Is the minus-sign stuff documented anywhere?  I feel like I tried to use it and failed at some point ... maybe 1.1.0 vs. master.)

**Review by richsalz — APPROVED** [review]
ah, thanks!

**Comment by levitte** [comment]
@kroeckx, I looked earlier and noticed that it's undocumented. I'm going to fix that in a separate PR. 
This is a master only feature... I think. I'll verify in an hour or so...

**Comment by levitte** [comment]
Misdirected comment, I meant @kaduk...

**Comment by levitte** [comment]
Confirmed just now that this truly is master only

**Comment by levitte** [comment]
Merged.  e5fd8ca43beeb40d33d421b519fb7e1d20358b94

**Comment by davidben** [comment]
Is it documented anywhere how running tests works, how to run individual ones, etc? I feel like every time I try to do it, it's a bit of an adventure. :-)

**Comment by levitte** [comment]
Actually, except for the numbers, it *is* documented.  Look for "TESTS in Detail" in `INSTALL`

**Comment by richsalz** [comment]
TL;DR:
```
make TESTS=test_ec V=1 tests
```
The ```V=1``` is optional, it means more verbose.
